src/service/vaults.ts

```diff
@@ -88,7 +95,14 @@ type VaultSubscribers = {
  * Ex. `asset: 'published.vaultFactory.manager0'` -> `'manager0'`.
  */
 const getManagerIdFromSubscribers = (subscribers: VaultSubscribers) =>
-  subscribers.vault.split('.').find(node => node.includes('manager'));
+  subscribers.vault
+    .split('.')
+    .find(node => node.startsWith('manager') && !node.startsWith('managers'));
```

Comment:
Given the example, `lastNode` would work but I take it this is used for longer paths like `published.vaultFactory.managers.manager0.metrics`. That would be good to include in the example.

Consider using a regex for further clarity. And refrain from "node" since that's the kind of the object for writing and this function is just string manipulation.
```suggestion
    .find(segment => segment.match(MANAGER_RE));
```
where with this in module scope:
```js
const MANAGER_RE = /manager\d+/;
```

Alternately (not recommended):
```suggestion
    .find(MANAGER_RE.test.bind(MANAGER_RE));
```

Reply:
Thanks for the regex, I normally try to avoid it for readability but this one seems simple enough.

>I take it this is used for longer paths like published.vaultFactory.managers.manager0.metrics

I was careful not to call it on that, only on the `prefix`, but moot point now.